### PR TITLE
LibGUI: Add an optional "automatic" autocomplete feature to TextEditor

### PR DESCRIPTION
**Pull request by alimpfard**
"automatic" autocomplete, where the editor shows suggestions unprompted (but tries to avoid being too obtrusive).

I briefly though about making only `<tab>` apply the highlighted suggestion in this mode, but it seemed weird (same for only `<return>` doing so)

Perhaps this should be the default?

### CODE REVIEW

File: Libraries/LibGUI/TextEditor.cpp [hunk 1]
```diff
@@ -1848,4 +1874,24 @@ void TextEditor::set_visualize_trailing_whitespace(bool enabled)
...
+        return;
+
+    if (value) {
+        ASSERT(m_autocomplete_provider);
+        m_autocomplete_timer = add<Core::Timer>(m_automatic_autocomplete_delay_ms, nullptr);
+        m_autocomplete_timer->set_single_shot(true);
+        m_autocomplete_timer->stop();
+        m_autocomplete_timer->on_timeout = [this] {
+            try_show_autocomplete();
+        };
```
Comment (awesomekling) [inline Libraries/LibGUI/TextEditor.cpp:1889]:
```suggestion
    m_autocomplete_timer = Core::Timer::create_single_shot(m_automatic_autocomplete_delay_ms, [this] { try_show_autocomplete(); });
```

**Comment by tomuta** [comment]
Slightly off-topic, but a LibSpell would be really cool. Would be nice for Browser, Spreadsheet, IRC client, and maybe even TextEditor...

File: Libraries/LibGUI/TextEditor.cpp (hunk 1 above)
Comment (alimpfard) [reply Libraries/LibGUI/TextEditor.cpp:1889]: That changed so incrementally I didn't even think of using that factory function...oops

**Comment by alimpfard** [comment]
> LibSpell

Where would we get the dictionary for that though?
We could generate it given some corpus, or just find one and grab it as a "port"?

It _would_ be nice, with this PR we'd even have autocorrect suggestions too

**Pull request merged**